File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/config/serializer/NotificationSerializer.java
package de.jalumu.magma.platform.base.config.serializer;

import de.exlll.configlib.Serializer;
import de.jalumu.magma.text.notification.Notification;

import java.util.HashMap;

public class NotificationSerializer implements Serializer<Notification, HashMap<String,Object>> {
    @Override
    public HashMap<String, Object> serialize(Notification element) {
        HashMap<String, Object> map = new HashMap<>();
        map.put("template",element.getNotificationTemplate());
        map.put("sound",element.getDefaultSound().name().namespace() + ":" + element.getDefaultSound().name().value());
        return map;
    }

    @Override
    public Notification deserialize(HashMap<String, Object> element) {
        return Notification.raw(element.get("template").toString(),element.get("sound").toString());
    }
}

File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/config/serializer/NotificationSerializer.kt
package de.jalumu.magma.platform.base.config.serializer

import de.exlll.configlib.Serializer
import de.jalumu.magma.text.notification.Notification

class NotificationSerializer : Serializer<Notification, HashMap<String, Any>> {
    override fun serialize(element: Notification): HashMap<String, Any> {
        val map = HashMap<String, Any>()
        map["template"] = element.notificationTemplate
        map["sound"] = element.defaultSound.name().namespace() + ":" + element.defaultSound.name().value()
        return map
    }

    override fun deserialize(element: HashMap<String, Any>): Notification {
        return Notification.raw(element["template"].toString(), element["sound"].toString())
    }
}
